src/core/thread/mle.cpp

```diff
@@ -4170,11 +4186,20 @@ const char *Mle::MessageTypeToString(MessageType aType)
 #endif
 #endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
 #else  // OPENTHREAD_FTD
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    static_assert(kTypeTimeSync == 16, "kTypeTimeSync value is incorrect");
```

Comment:
I think we can clean this up a bit by moving `kTypeTimeSync` to the end of the `MessageType` enum definition. Thoughts?

Reply:
I can do it but it will change value of kTypeLinkMetricsManagementRequest, kTypeLinkMetricsManagementResponse, kTypeLinkProbe. Just tell me if is fine to do it so.

Reply:
I think that should be fine - these enums are primarily used for logging.

Reply:
Done